src/test/groovy/netflix/nebula/dependency/recommender/DependencyRecommendationsPluginCoreBomSupportSpec.groovy

```diff
@@ -0,0 +1,108 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package netflix.nebula.dependency.recommender
+
+import nebula.test.IntegrationSpec
+import nebula.test.dependencies.DependencyGraphBuilder
+import nebula.test.dependencies.GradleDependencyGenerator
+import nebula.test.dependencies.maven.ArtifactType
+import nebula.test.dependencies.maven.Pom
+import nebula.test.dependencies.repositories.MavenRepo
+import spock.lang.Unroll
+
+class DependencyRecommendationsPluginCoreBomSupportSpec extends IntegrationSpec {
+    def repo
+    def generator
+
+    def setup() {
+        fork = true
+        new File("${projectDir}/gradle.properties").text = "systemProp.nebula.features.coreBomSupport=true"
+
+        settingsFile << """\
+            enableFeaturePreview('IMPROVED_POM_SUPPORT')
+            """.stripIndent()
+
+        repo = new MavenRepo()
+        repo.root = new File(projectDir, 'build/bomrepo')
+        def pom = new Pom('test.nebula.bom', 'testbom', '1.0.0', ArtifactType.POM)
+        pom.addManagementDependency('test.nebula', 'foo', '1.0.0')
+        repo.poms.add(pom)
+        repo.generate()
+        def graph = new DependencyGraphBuilder()
+                .addModule('test.nebula:foo:1.0.0')
+                .build()
+        generator = new GradleDependencyGenerator(graph)
+        generator.generateTestMavenRepo()
+    }
+
+    def 'add given bom to configs'() {
+        buildFile << """\
+            apply plugin: 'nebula.dependency-recommender'
+            apply plugin: 'java'
+
+            repositories {
+                maven { url '${repo.root.absoluteFile.toURI()}' }
+                ${generator.mavenRepositoryBlock}
+            }
+
+            dependencyRecommendations {
+                mavenBom module: 'test.nebula.bom:testbom:latest.release'
+            }
+
+            dependencies {
+                compile 'test.nebula:foo'
+            }
+            """.stripIndent()
+
+        when:
+        def result = runTasksSuccessfully('dependencies', '--configuration', 'compileClasspath')
+
+        then:
+        println result.standardOutput
+        result.standardOutput.contains("+--- test.nebula:foo -> 1.0.0")
+    }
+
+    @Unroll
+    def 'error when #type(#argType) used'() {
+        given:
+        buildFile << """\
+            apply plugin: 'nebula.dependency-recommender'
+            apply plugin: 'java'
+
+            dependencyRecommendations {
+                $type $arg
+            }
+            """.stripIndent()
+
+        when:
+        def result = runTasksWithFailure('dependencies', '--configuration', 'compileClasspath')
+
+        then:
+        result.standardOutput.contains("> dependencyRecommender.$type is not available with 'systemProp.nebula.features.coreBomSupport=true'")
+
+        where:
+        type             | argType   | arg
```

Comment:
Alas, core Gradle bom support won't support these?

Reply:
Correct, it's only about improving gradle's pom support